rfcs/0036-rfc-process-team-amendment.md

```diff
@@ -0,0 +1,202 @@
+---
+feature: rfc-process-team-amendment
+start-date: 2018-10-27
+author: Robin Gloster <mail@glob.in>
+co-authors: Graham Christensen <graham@grahamc.com>
+related-issues: 1 (initial process), 24 (implementation)
+---
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an RFC Steering Committee who decide on a group of RFC
+shepherds for each RFC who guide the discussion to a general consensus and then
+propose a motion for a "Final Comment Period" (FCP) with a disposition for
+acception, rejection or postponing. (see Glossary for a short definition of
+these terms)
+
+
+# Motivation
+[motivation]: #motivation
+
+A lot of RFCs have stalled and already an [RFC has been submitted exactly on
+this topic](https://github.com/NixOS/rfcs/pull/18), which ironically has not
+been decided on either. This new RFC takes the above into account and tries to
+expand on that to flesh out the process further. During this effort a lot of
+inspiration has been taken from [Rust's RFC
+process](https://github.com/rust-lang/rfcs#what-the-process-is) which works well
+and we have adapted to our needs.
+
+
+# Detailed design
+[design]: #detailed-design
+
+## Glossary
+
+##### RFC Steering Committee
+A team of people defined by _this_ RFC and stays consistent until the team
+members are changed via a follow-up RFC. This committee is responsible for
+forming an RFC Shepherd team from the available nominations on each RFC. This
+team also names the leader of the Shepherd team. This has to happen within 1
+week after the PR has been opened. Until then the Steering Committee is
+responsible for guiding the discussion. In case of the Shepherding Team not
+doing its work the Steering Committee shall encourage them or step in and assign
+new Shepherds. They also are in charge of merging accepted and rejected RFCs.
+Generally by these expectations they should find time to meet once a week for
+about an hour.
+
+##### Shepherd
+A team of 3-4 community members defined unanimously by the RFC Steering
+Committee, responsible for accepting or rejecting RFCs. This team is created per
+RFC from community members nominated in the discussion on that RFC.
+
+This team should be people who are very familiar with the main components
+touched by the RFC.
+
+##### Leader
+The Shepherd Leader is in charge of the RFC process for a specific RFC, and is
+responsible for ensuring the process is followed in a timely fashion. 
+
+
+## Process from Creation to Merge
+
+*In short, to get a major change included in Nix or nixpkgs, one must
+first get the RFC merged into the RFC repository as a markdown file under the
+`accepted` directory. At that point the RFC is accepted and may be implemented
+with the goal of eventual inclusion into Nix or nixpkgs.*
+
+0. Have a cool idea!
+1. Find a co-author. A co-author is critical to making sure your RFC is viable
+   and will receive support. Your co-author helps flesh out the RFC, and should
+   also support the RFC.
```

Comment:
One can still write up the initial draft on one's own, but then before PR'ing a co-author should be found, who will probably have some remarks to flesh it out further.

Reply:
That's true; but watching RFC repository was previously a reasonable way to notice the discussion on still-raw RFCs. Now it will spill in all the different directions…

Reply:
I don't have a strong opinion here, but FWIW I've never quite understood why a co-author is critical nor, in the case where there is clearly a single driving force, what the co-author is actually expected to do.

Reply:
I also don’t quite understand the motivation behind the co-author. Is it just to signal that more than a single person believe the suggestion is a good idea? Or is it more about increasing the quality of the initial draft by having a second pair of eyes looking over it before a bigger audience gets to review it? 

Depending on the actual motivation I wonder if it might end up being just a formality without too much added value?

Reply:
The co-author was introduced in the initial RFC process to play some of the role of the sheperd's team; help shape the RFC and clean it before submitting the RFC for wider consumption. There was also the idea that if the author can't find at least one co-author then it will be hard motivating more people that the RFC is worth considering.

Reply:
Please vote with thumbs on this comment: remove the co-author from the RFC

Reply:
@grahamc and I decided to keep it mainly for the reason of keeping up the standard of opened PRs. For poorly written our thought out RFCs that probably won't be accepted, it would add unnecessary work for the RFC SC. A second set of eyes would hopefully ensure that there is support of at least a few people and obvious fallacies can be addressed before a larger audience is let loose on it.

Reply:
An idea: replace requirement for the co-author with "find at least one person that agrees to be nominated as part of the shepherding Team"?

So you have confirmation that at least one other person considers this important. Also, a possible shepherding member is usually more experienced so can give advice during rfc draft process.

Reply:
Removed the necessity for a co-author and fleshed out the request to the author, to gather feedback at an early stage/before opening the PR.